### PR TITLE
[skip ci] Added auth chart

### PR DESCRIPTION
**Pull request by suvajit-sarkar**
Signed-off-by: suvajit-sarkar <suvajit.sarkar@accenture.com>

**Changelog**
- Add helm chart for auth service
- Update pki job to create auth trust store

**TODO**
Creds for the crypto materials needs to fetch vault and replaced
JWT creation can be moved to PKI job

 

**Reviewed by**
@sownak @jagpreetsinghsasan 

 

**Linked issue**
#1506

### CODE REVIEW

File: platforms/r3-corda-ent/charts/auth/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.prefix }}-auth-etc
+  namespace: {{ .Values.metadata.namespace }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storage.nameName: {{ .Values.storage.name }}
```
Comment (surabhi-dudhediya) [inline platforms/r3-corda-ent/charts/auth/templates/pvc.yaml:16]: storage.nameName should be storageClassName

File: platforms/r3-corda-ent/charts/auth/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.prefix }}-auth-h2
+  namespace: {{ .Values.metadata.namespace }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storage.nameName: {{ .Values.storage.name }}
```
Comment (surabhi-dudhediya) [inline platforms/r3-corda-ent/charts/auth/templates/pvc.yaml:30]: storage.nameName should be storageClassName

File: platforms/r3-corda-ent/charts/auth/templates/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.prefix }}-auth-logs
+  namespace: {{ .Values.metadata.namespace }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storage.nameName: {{ .Values.storage.name }}
```
Comment (surabhi-dudhediya) [inline platforms/r3-corda-ent/charts/auth/templates/pvc.yaml:44]: storage.nameName should be storageClassName

File: platforms/r3-corda-ent/charts/auth/templates/pvc.yaml (hunk 1 above)
Comment (suvajit-sarkar) [reply platforms/r3-corda-ent/charts/auth/templates/pvc.yaml]: fixed

File: platforms/r3-corda-ent/charts/auth/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{{- define "labels.custom" -}}
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/auth/templates/_helpers.tpl:1]: labels.custom never used

File: platforms/r3-corda-ent/charts/auth/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: vault-auth
+      securityContext:
+        {{- toYaml .Values.config.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: init-certificates
+          image: {{ .Values.image.initContainerName }}
+          imagePullPolicy: Always
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/auth/templates/deployment.yaml:42]: wrong pull policy

**Review by sownak — CHANGES_REQUESTED** [review]

**Review by sownak — APPROVED** [review]

**Pull request merged**